src/NATS.Client.Core/Internal/ClientOpts.cs

```diff
@@ -97,23 +97,12 @@ public static ClientOpts Create(NatsOpts opts)
 
     private static string GetAssemblyVersion()
     {
-        var asm = typeof(ClientOpts);
-        var version = "1.0.0";
-        var infoVersion = asm!.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
-        if (infoVersion != null)
-        {
-            version = infoVersion.InformationalVersion;
-        }
-        else
-        {
-            var asmVersion = asm!.GetCustomAttribute<AssemblyVersionAttribute>();
-            if (asmVersion != null)
-            {
-                version = asmVersion.Version;
-            }
-        }
-
-        return version;
+        var asm = typeof(ClientOpts).Assembly;
+
+        return asm?.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version
+            ?? asm?.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion
```

Comment:
Question about which attribute to use:

Just playing with it changing version.txt here are the differences between file and informational versions:

| version.txt     | file    | informational                                                   |
|-----------------|---------|-----------------------------------------------------------------|
| 2.3.0           | 2.3.0.0 | 2.3.0\n\u002B040efe2cffc07cbbcb4267afdc779275aa12d4a7           |
| 2.3.0-preview.1 | 2.3.0.0 | 2.3.0-preview.1\n\u002B040efe2cffc07cbbcb4267afdc779275aa12d4a7 |

do you think we should chop informational so that we can send preview suffixes as well?

Reply:
In that case, let's just use the first part of the informational attribute, and forget trying the other ones.  I don't see the point in trying for the other attributes....

So just a simplied version;

```
   var asm = typeof(ClientOpts).Assembly;
   var version = "2.0.0";

   var attrInformationalVersion = asm!.GetCustomAttribute<AssemblyInformationalVersionAttribute>();
   if (attrInformationalVersion != null)
   {
       version = attrInformationalVersion.InformationalVersion.Split('\n')[0];
   }

   return version;
```